VideoLocker/src/main/java/org/edx/mobile/module/prefs/UserPrefs.java

```diff
@@ -51,20 +52,21 @@ public boolean isDownloadOverWifiOnly() {
      */
     public File getDownloadFolder() {
         ProfileModel profile = getProfile();
-        
-        File android = new File(Environment.getExternalStorageDirectory(), "Android");
-        File downloadsDir = new File(android, "data");
-        File packDir = new File(downloadsDir, context.getPackageName());
-        File edxDir = new File(packDir, profile.username);
-        edxDir.mkdirs();
+        File videosDir = new File(context.getExternalFilesDir(null).getParent(), AppConstants.Folders.VIDEOS);
```

Comment:
> As for how to handle the upgrade, we also need to account for the case where we only did part of the migration, as the migration is non-transactional.

Hmm, so thinking about this, we could actually make this transactional, by performing the whole upgrade in a transaction, and then only marking it as successful after the directory is successfully renamed (which is an atomic operation as well). Otherwise, the whole transaction would be rolled back. If the external files directory path is inaccessible from the start, then the transaction would be aborted as well.

Note that `SQLiteOpenHelper` already performs the upgrade inside a transaction. The Android SQLite framework allows nested transactions, which are actually performed as one single transaction. If any nested transaction is unsuccessful, then the root transaction is rolled back. This should have the implication the database should be accessible in the original version (since the version upgrade would be rolled back as well) and schema until the migration is completed. This would have one caveat though, which is that if we're upgrading from an even older version, then the upgrade wouldn't be performed at all, since it's all done in one transaction. It's been more than a year since the last database update though, so probably there aren't many people at these old versions who would be updating now.